src/System.Net.Primitives/src/System/Net/IPEndPoint.cs

```diff
@@ -119,6 +119,68 @@ public int Port
             }
         }
 
+        public static bool TryParse(string s, out IPEndPoint result)
+        {
+            return TryParse(s.AsSpan(), out result);
+        }
+
+        public static bool TryParse(ReadOnlySpan<char> s, out IPEndPoint result)
+        {   
+            int addressLength = s.Length;  // If there's no port then send the entire string to the address parser
+            int lastColonPos = s.LastIndexOf(':');
+
+            // Look to see if this is an IPv6 address with a port.
+            if (lastColonPos > 0)
+            {
+                if (s[lastColonPos - 1] == ']')
+                {
+                    addressLength = lastColonPos;
+                }
+                // Look to see if this is IPv4 with a port (IPv6 will have another colon)
+                else if (s.Slice(0, lastColonPos).LastIndexOf(':') == -1)
+                {
+                    addressLength = lastColonPos;
+                }
+            }
+
+            if (IPAddress.TryParse(s.Slice(0, addressLength), out IPAddress address))
+            {
+                int port = 0;
+                if (
+                    (addressLength == s.Length || int.TryParse(s.Slice(addressLength + 1), out port)) &&
+                    port >= MinPort && port <= MaxPort
+                    )
+                    
+                {
```

Comment:
This is grouped incorrectly.... the port range check is only necessary if we did TryParse... it's not necessary if addressLength == s.Length. (The formatting is also different from what we use elsewhere in corefx.)  I'd have expected this to be:
```C#
int port = 0;
if (addressLength == s.Length ||
    (int.TryParse(s.Slice(addressLength + 1), out port) && port >= MinPort && port <= MaxPort))
{
 ...
}
```